lib/plugins/aws/package/lib/mergeIamTemplates.js

```diff
@@ -66,7 +73,14 @@ module.exports = {
     iamRoleLambdaExecutionTemplate.Properties.Path = this.provider.naming.getRolePath();
     iamRoleLambdaExecutionTemplate.Properties.RoleName = this.provider.naming.getRoleName();
 
-    if (this.serverless.service.provider.rolePermissionsBoundary) {
+    // set permission boundary
+    if (iamRole.permissionBoundary) {
+      iamRoleLambdaExecutionTemplate.Properties.PermissionsBoundary = iamRole.permissionBoundary;
+    } else if (this.serverless.service.provider.rolePermissionsBoundary) {
+      this.serverless._logDeprecation(
```

Comment:
Deprecations should be configured at `initialize` hook (as e.g. here: https://github.com/serverless/serverless/blob/12805c3d152d85af9dba3dd3ecfa2002a621f6a8/lib/plugins/aws/package/compile/functions.js#L31-L61), as that guarantees that deprecation are shown at early stage of command processing and even if affected functionality is not used (e.g. we do not package service, or via some other logic processing of those properties is skipped).

It's important as with next major we will throw on not supported properties, so deprecation of existing properties needs to be communicated in case of all commands.